npc/custom/item_options.txt

```diff
@@ -0,0 +1,234 @@
+//=
+//= This program is distributed in the hope that it will be useful,
+//= but WITHOUT ANY WARRANTY; without even the implied warranty of
+//= MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//= GNU General Public License for more details.
+//=
+//= You should have received a copy of the GNU General Public License
+//= along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//=========================================================================
+//= Option Master (Item Options NPC @see db/item_options.conf)
+//================= Description ===========================================
+//= Select from a list of your equipments, one to enhance with up to 5 options.
+//= A list of the first 10 options are provided in this npc.
+//================= Current Version =======================================
+//= 1.0
+//==================== Credits ============================================
+//= Smokexyz
+//=========================================================================
+prontera,153,180,4	script	Option Master	4_DOG01,{
+	callsub(StartTalking);
+	end;
+	
+OnInit:
+	/**
+	 * General Configuration
+	 */
+	/* Chance of the enhancement process to fail. (0 - 99 in percent) */
+	.chance_of_failure = 10;
+	/* Delete the item on failure? (true/false) */
+	.delete_on_failure = true;
+	/* Required amount of zeny for a try. */
+	.zeny_requirement = 100;
+	/* Minimum amount of the bonus value. 
+	 * For negative effects or certain bonuses that require negative values
+	 * Maximum possible value is -INT16_MAX)
+	 */
+	.minimum_bonus_amount = -100;
+	/* Maximum amount of the bonus value. 
+	 * Maximum possible value is INT16_MAX */
+	.maximum_bonus_amount = 100;
+	/* Disable selection of bonus value (true/false) */
+	.enable_random_bonus = false;
+	
+	/* Item Option Descriptions */
+	setarray .options$[0],"Max HP", "Max SP", "STR", "AGI", "VIT", "INT", "DEX", "LUK";
+	/* Item Option Constants */
+	setarray .option_constants[0], VAR_MAXHPAMOUNT, VAR_MAXSPAMOUNT, VAR_STRAMOUNT, VAR_AGIAMOUNT, VAR_VITAMOUNT, VAR_INTAMOUNT, VAR_DEXAMOUNT, VAR_LUKAMOUNT;
+	end;
+	
+StartTalking:
+	.@name$ = "[ ^990099Option Master^000000 ]";
+	disable_items;
+	mes(.@name$);
+	mes("Hey there "+strcharinfo(0));
+	mes("There's a new-found method of adding options to an equipment that I have been working on.");
+	next;
+	if (.enable_random_bonus) {
+		mes(.@name$);
+		mes("Equipment bonuses are added randomly per slot!");
+		mes("So if you wind up with a low amount of bonus, don't hesitate to override the slot.");
+		next;
+	}
+	mes(.@name$);
+	mes("Would you like me to add extra stats to your equipment?");
+	next;
+	if (select("Of course!", "Meh, not right now.") == 2) {
+		mes(.@name$);
+		mes("Alright then, catch you later.");
+		close;
+	}
+	// Build the Menu.
+	setarray .@position$[1],"Head","Body","Left Hand","Right Hand","Robe","Shoes","Accessory 1","Accessory 2","Head 2","Head 3";
+	.@menu$ = "";
+	for (.@i = 1; .@i<=10; ++.@i)
```

Comment:
```
EQI_HEAD_TOP (1)          - Upper head gear
EQI_ARMOR (2)             - Armor (Where you keep your Jackets and Robes)
EQI_HAND_L (3)            - What is in your Left hand.
EQI_HAND_R (4)            - What is in your Right hand.
EQI_GARMENT (5)           - The garment slot (Mufflers, Hoods, Manteaus)
EQI_SHOES (6)             - What foot gear the player has on.
EQI_ACC_L (7)             - Accessory 1.
EQI_ACC_R (8)             - Accessory 2.
EQI_HEAD_MID (9)          - Middle Headgear (masks and glasses)
EQI_HEAD_LOW (10)         - Lower Headgear (beards, some masks)
```
btw, if you're referring to this list... to use the number 1 ~ 10 ... it's better to replace with the `EQI_` constant.
it could help avoid potential script broken if these EQI values changed in future..... just how what happened in rA when they decided to update the values.